### PR TITLE
README: Update and enhance the readme file

### PR DESCRIPTION
**Pull request by yosefe**
## Why
- Improve the usability/organization of README file
- Add link to read-the-docs website

--
View the updated README file: https://github.com/openucx/ucx/blob/a482093ecb65db2b2c69282583fb81211e7a6e2b/README.md

### CODE REVIEW

**Comment by yosefe** [comment]
View the updated READ file: https://github.com/openucx/ucx/blob/3ef03a61c6946c66cded15f587247ba85f867055/README.md

**Comment by shamisp** [comment]
@yosefe can you please split the changes to two parts ? First - just the README content update. Second - moving from one file to another. Otherwise it is very difficult to review.

**Comment by yosefe** [comment]
@shamisp reverted symlink change, will be another PR

File: README [hunk 1]
```diff
@@ -32,25 +16,30 @@ layer for Message Passing ([MPI](https://www.mpi-forum.org/)),
 [PGAS](http://www.pgas.org/)/[OpenSHMEM](http://www.openshmem.org/)
```
Comment (shamisp) [inline README:16]: optimized communication layer -> optimized communication framework.

I also don't think the above reflects up to date scope of UCX. We need to work a bit more on this definition. It does not really reflect ML, AI, and data analytics, etc.

Comment (yosefe) [reply README:16]: @shamisp the wording has not changed in this PR.
WDYT about taking it from RTD:
https://openucx.readthedocs.io/en/master/#openucx

Comment (shamisp) [reply README:16]: IMHO RTD is better

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
bot:pipe:retest

File: README (hunk 1 above)
Comment (yosefe) [reply README]: @shamisp fixed, can u pls look?

Comment (shamisp) [reply README]: Look good. @tonycurtis can you please take a look at the PR.

File: README [hunk 2]
```diff
@@ -7,50 +7,40 @@
...
+# Unified Communication X
 
-<hr>
+Unified Communication X (UCX) is an
+[award winning](https://losalamosreporter.com/2019/11/07/nine-los-alamos-national-laboratory-projects-win-rd-100-awards),
+optimized production proven communication framework for modern, high-bandwidth
+and low-latency networks.
 
-# Unified Communication X
+UCX exposes a set of abstract communication primitives which utilize the best of
```
Comment (tonycurtis) [inline README:19]: which -> that

Memory -> memory

File: README [hunk 3]
```diff
@@ -7,50 +7,40 @@
...
-      * [UCX Publications](#ucx-publications)
-      * [UCX Architecture](#ucx-architecture)
-      * [Supported Transports](#supported-transports)
-      * [Supported CPU Architectures](#supported-cpu-architectures)
+# Unified Communication X
 
-<hr>
+Unified Communication X (UCX) is an
+[award winning](https://losalamosreporter.com/2019/11/07/nine-los-alamos-national-laboratory-projects-win-rd-100-awards),
+optimized production proven communication framework for modern, high-bandwidth
```
Comment (tonycurtis) [inline README:16]: production-proven

File: README [hunk 4]
```diff
@@ -112,13 +96,83 @@ Start server:
...
 $ ./src/tools/perf/ucx_perftest -c 0
 ```
-
 Connect client:
 
 ```sh
 $ ./src/tools/perf/ucx_perftest <server-hostname> -t tag_lat -c 1
 ```
-Note: the `-c` flag sets CPU affinity. If running both commands on same host, make sure you set the affinity to different CPU cores.
+> NOTE the `-c` flag sets CPU affinity. If running both >commands on same host, make sure you set the affinity to different CPU cores.
```
Comment (tonycurtis) [inline README:104]: Stray ">" before "commands"

**Review by tonycurtis — CHANGES_REQUESTED** [review]

File: README (hunk 4 above)
Comment (yosefe) [reply README:104]: this add indentation

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
bot:pipe:retest

File: README (hunk 4 above)
Comment (yosefe) [reply README:104]: @tonycurtis fixed

**Review by tonycurtis — APPROVED** [review]

**Pull request merged**